### PR TITLE
[DVC-5123] android store read anon user id on initialize

### PR DESCRIPTION
**Pull request by suthar26**
when initializing the DVCClient, compare and store anonymous user id to SharedPrefs to not create multiple anon users for same device.

### CODE REVIEW

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/model/User.kt
Comment (ajwootto) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/model/User.kt:22]: this should remain immutable, a user object shouldn't have its userId modified. Any change to userId should always require a new user object (this helps prevented unwanted side effects of various bits of code unexpectedly mutating the currently active user instance)

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt
Comment (suthar26) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt:401]: could move this to `SharedPrefs` to not have duplicate code

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt
Comment (suthar26) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt:516]: could move this to `SharedPrefs` to not have duplicate code

Comment (laurawarr) [reply android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt:507]: these wrappers aren't doing much, we could just use the dvcSharedPrefs getString/saveString methods directly, or move them into DVCSharedPrefs — your call

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/model/User.kt
Comment (laurawarr) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/model/User.kt:87]: the nested ternary on a single line doesn't make this very readable, maybe we could split it up?
```suggestion
            val userId = if (isAnonymous) {
                if (anonUserId == null) UUID.randomUUID().toString() else anonUserId
            } else {
                user.userId
            }
```

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/util/DVCSharedPrefs.kt
Comment (laurawarr) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/util/DVCSharedPrefs.kt:74]: considering we'll be using this even when it's okay for the value to be null, maybe we should switch this to a debug/info log rather than an error (I believe that'd just Timber.d or Timber.i)

**Comment by laurawarr** [comment]
Based on the failing tests, it looks like you missed one reference to `setAnonUserId` when you removed that method.

The code changes look — for future commits I'd try to keep the message relevant to the changes, once this PR is merged "reset and squash changes" won't have much meaning to other developers looking at the commit history

**Review by laurawarr — APPROVED** [review]

**Pull request merged**
